openmdao/core/group.py

```diff
@@ -2598,102 +2597,31 @@ def _get_auto_ivc_out_val(self, tgts, remote_vars, all_abs2meta, abs2meta):
-                                       "src_indices are not supported.")
-                maxends.append(np.max([end for _, end, _ in ranges]))
-                sizes.append(np.sum([sz for _, _, sz in ranges]))
-                fulls.append(np.all([sz == (end - start + 1) for start, end, sz in ranges]))
-
-            is_default = (np.all(fulls) and np.all([m == 0 for m in minstarts])
-                          and np.all([m + 1 == sizes[0] or m == 0 for m in maxends]))
-
-            if is_default:  # all src_indices are non-overlapping starting at 0
-                # just match local size of auto_ivc to local size of input(s)
-
-                # must check that values for different distrib inputs match
-                val = None
-                old_tgt = None
-                for tgt, _ in dist_ranges:
-                    if tgt in abs2meta:
-                        if val is None:
-                            val = abs2meta[tgt]['value']
-                            old_tgt = tgt
-                        elif np.linalg.norm(val - abs2meta[tgt]['value']) > 1e-40:
-                            raise RuntimeError(f"{self.msginfo}: Can't connect "
-                                               "_auto_ivc to distributed variables "
-                                               f"'{tgt}' and '{old_tgt}' because they "
-                                               "have different values in rank "
-                                               f"{self.comm.rank}.")
-
-                if val is None:
-                    val = np.zeros(0)
-            elif len(dist_ranges) > 1:
-                raise RuntimeError(f"{self.msginfo}: Can't connect _auto_ivc to distributed"
-                                   f" inputs {sorted(t[0] for t in dist_ranges)}.")
-            else:
-                # distributed size layout is undefined, so just take the total distrib
-                # size and chop it up evenly among procs where the input size > 0.
-                num_nz_divisions = len([t for t, rngs in dist_ranges if rngs[1][2] > 0])
-                total_size = np.max(maxends)
-                sizes, _ = evenly_distrib_idxs(num_nz_divisions, total_size)
-                tgt, ranges = dist_ranges[0]
-                nz_count = 0
-                for rank, (start, end, sz) in enumerate(ranges):
-                    if rank == myrank:
-                        val = np.ones(sizes[nz_count])
-                        break
-                    if sz > 0:
-                        nz_count += 1
-
-            info.append((tgt, val.size, val, True))
+        if src_idx_found:  # auto_ivc connected to local vars with src_indices
+            tgts = ', '.join(src_idx_found)
+            msg = 'The following inputs [{}] are defined using src_indices. These cannot be '
+            msg += 'connected to an automatic IndepVarComp because the source size is '
+            msg += 'undetermined.  Please add one manually and connect it.'
```

Comment:
I think that the whole "AutoIVC thing is supposed to be transparent to the user?
May be better to just say that an IndepVarComp is required to determine the source size?

Reply:
Simplified both messages.